include/eve/module/core/regular/min.hpp

```diff
@@ -12,6 +12,30 @@
 
 namespace eve
 {
+  template<typename Options>
+  struct min_t : tuple_callable<min_t, Options, pedantic_option, numeric_option>
+  {
+    template<eve::ordered_value T, ordered_value U>
+    EVE_FORCEINLINE constexpr common_value_t<T, U> operator()(T t, U u) const noexcept { return EVE_DISPATCH_CALL(t, u); }
+
+    template<eve::ordered_value T0, ordered_value T1, ordered_value... Ts>
+    EVE_FORCEINLINE constexpr common_value_t<T0, T1, Ts...> operator()(T0 t0, T1 t1, Ts...ts) const noexcept
+    {
+      return EVE_DISPATCH_CALL(t0,  t1, ts...);
+    }
+
+    template<kumi::non_empty_product_type Tup>
+    EVE_FORCEINLINE constexpr kumi::apply_traits_t<eve::common_value,Tup>
+    operator()(Tup const & t) const noexcept  requires(kumi::size_v<Tup> >= 2) { return EVE_DISPATCH_CALL(t); }
+
+    template<typename Callable>
+    requires(!kumi::product_type<Callable> && !eve::ordered_value<Callable>)
+    EVE_FORCEINLINE constexpr auto operator()(Callable const & f) const noexcept { return EVE_DISPATCH_CALL(f); }
```

Comment:
@jfalcou do we want to change the return type here from auto, as w discussed?

Reply:
Will do in a next PR